Library/Homebrew/livecheck/strategy/bitbucket.rb

```diff
@@ -46,30 +53,23 @@ def self.match?(url)
         # @param regex [Regexp] a regex used for matching versions in content
         # @return [Hash]
         def self.find_versions(url, regex = nil)
-          %r{
-            bitbucket\.org/
-            (?<path>.+?)/ # The path leading up to the get or downloads part
-            (?<dl_type>get|downloads)/ # An indicator of the file download type
-            (?<prefix>(?:[^/]+?[_-])?) # Filename text before the version
-            v?\d+(?:\.\d+)+ # The numeric version
-            (?<suffix>[^/]+) # Filename text after the version
-          }ix =~ url
+          match = url.match(URL_MATCH_REGEX)
 
           # Use `\.t` instead of specific tarball extensions (e.g. .tar.gz)
-          suffix.sub!(/\.t(?:ar\..+|[a-z0-9]+)$/i, "\.t")
+          match[:suffix].sub!(/\.t(?:ar\..+|[a-z0-9]+)$/i, "\.t")
```

Comment:
> this doesn't seem to be related to supporting casks

There was at least one cask which was wrongfully using this strategy because the `URL_MATCH_REGEX` here is too broad. I guess it's not an issue anymore since `strategy :sparkle` is now specified explicitly in all casks.

Reply:
If I understand this correctly, it sounds like both `Bitbucket` and `Sparkle` matched the URL (when `Sparkle was applied automatically previously) and `Bitbucket` was being applied due to having the higher priority when you wanted to use `Sparkle`.

If that's the case, then specifying the strategy in the `livecheck` block was the appropriate fix in this situation. We have formulae that do this in homebrew/core with `strategy :page_match` when a more specific strategy applies to a given URL (e.g., using `PageMatch` to check a SourceForge page instead of checking the RSS feed using `Sourceforge`).

---

I agree that the existing `URL_MATCH_REGEX` in `Bitbucket` isn't properly enforcing the URL requirements (due to being too loose). I also agree that using the same regex in `#match?` and `#find_versions` is a good idea in a general sense, though it's still frustrating that we can only use `=~` with a regex literal. [I worked on this same refactoring in the past and didn't find a solution for addressing the immutability of named capture group values when using `#match`.]

That said, I think the benefit of using the regex in both places outweighs the detriment of the named capture group values being immutable and we'll be going in this direction anyway. Since this addressed an issue with a cask at one point, I'm fine leaving the `Bitbucket` changes in and I'll just bring the other strategies in line in a follow-up PR afterward.